### PR TITLE
Modify test usecase and mock repo for test

### PR DESCRIPTION
**Pull request by Filldexnoi**
Use the mock library to facilitate easier creation of mocks.

### CODE REVIEW

File: Handler/handler_product.go [hunk 1]
```diff
@@ -60,21 +61,23 @@ func (h *ProductHandler) UpdateProduct(c *fiber.Ctx) error {
...
 	}
-	err = h.UseCase.DeleteProduct(uint(id))
+	productEntity, err := h.UseCase.DeleteProduct(uint(id))
 	if err != nil {
-		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"error": err.Error()})
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": err.Error()})
 	}
-	return c.SendStatus(fiber.StatusNoContent)
+	ResProduct := payload.ProductToRespondProduct(productEntity)
+	return c.JSON(ResProduct)
```
Comment (kimzey) [inline Handler/handler_product.go]: last return in func handler no status ?

c.JSON() > c.Status().JSON()

File: Usecase/usecase_transaction_test.go [hunk 2]
```diff
@@ -1,157 +1,129 @@
...
-				Items:        []entities.Item{{ProductId: 1, Quantity: 5}, {ProductId: 2, Quantity: 3}},
-			},
-		)
+		mockProductRepo.On("GetPriceProducts", transaction).Return(transaction, nil).Once()
+		mockTransactionRepo.On("SaveCreateTransaction", transaction).Return(transaction, nil).Once()
+		result, err := service.CreateTransaction(transaction)
 		assert.NoError(t, err)
+		assert.NotNil(t, result)
+		assert.Equal(t, transaction, result)
+		mockTransactionRepo.AssertExpectations(t)
```
Comment (kimzey) [inline Usecase/usecase_transaction_test.go:51]: You dont have mockProductRepo.AssertExpectations(t) ?

File: Usecase/usecase_transaction_test.go [hunk 3]
```diff
@@ -1,157 +1,129 @@
...
-		err := service.CreateTransaction(&entities.Transaction{
-			OrderAddress: "th",
-			Items:        []entities.Item{{ProductId: 1, Quantity: 5}, {ProductId: 2, Quantity: 3}},
-		})
+		mockProductRepo.On("GetPriceProducts", transaction).Return(nil, errors.New("cannot get price products")).Once()
+		result, err := service.CreateTransaction(transaction)
 		assert.Error(t, err)
+		assert.Nil(t, result)
 		assert.Equal(t, "cannot get price products", err.Error())
+		mockTransactionRepo.AssertExpectations(t)
```
Comment (kimzey) [inline Usecase/usecase_transaction_test.go]: You dont have mockProductRepo.AssertExpectations(t) ?
why use  mockTransactionRepo.AssertExpectations(t), this case not use mockTransactionRepo.on()

File: Repo/product.go [hunk 4]
```diff
@@ -16,24 +15,38 @@ func NewProductRepo(db *gorm.DB) ProductRepoI {
...
-	err := r.db.Create(gormProduct).Error
+func (r *ProductDB) SaveCreateProduct(product *entities.Product) (*entities.Product, error) {
+	createProduct := models.ProductToGormProduct(product)
+	err := r.db.Create(&createProduct).Error
 	if err != nil {
-		return errors.New("cannot create product")
+		return nil, err
 	}
-	return nil
+	productEntity := createProduct.ToProduct()
```
Comment (engKIKO) [inline Repo/product.go:24]: Use **SaveGetProduct()** function in the SaveCreateProduct() or in productUseCase for get product and verify Product is added. and you can do this with Create and Update method on Stock, Order, transaction,and more

**Review by engKIKO — APPROVED** [review]

File: Repo/product.go (hunk 4 above)
Comment (Filldexnoi) [reply Repo/product.go:24]: The createProduct variable, when using r.db.create, sends the created data to the pointer &createProduct. This means that createProduct holds the data created in the database and can be returned for verification without needing to use the SaveGetProduct command to fetch the data for checking.

File: Usecase/usecase_transaction_test.go (hunk 3 above)
Comment (Filldexnoi) [reply Usecase/usecase_transaction_test.go]: mockProductRepo.On("GetPriceProducts", transaction).Return(nil, errors.New("cannot get price products")).Once() this one?

File: Usecase/usecase_transaction_test.go (hunk 2 above)
Comment (Filldexnoi) [reply Usecase/usecase_transaction_test.go:51]: what do you mean?

File: Handler/handler_product.go (hunk 1 above)
Comment (Filldexnoi) [reply Handler/handler_product.go]: Fixed it. Please check again.

File: Usecase/usecase_transaction_test.go (hunk 3 above)
Comment (Filldexnoi) [reply Usecase/usecase_transaction_test.go]: Fixed it. Please check again.

**Pull request merged**